backend/src/db.rs

```diff
@@ -0,0 +1,32 @@
+//! Database related things.
+
+use anyhow::Result;
+use deadpool_postgres::{Config, Pool};
+use log::{debug, info, trace};
+use tokio_postgres::NoTls;
+
+use crate::config;
+
+
+/// Creates a new database connection pool.
+pub async fn create_pool(config: &config::Db) -> Result<Pool> {
+    let config = Config {
```

Comment:
Can I convince you to rename things to make it

```
let pool_config = PoolConfig {
```

I find it slightly confusing, that there is the module `config`, which is our config, but then there is also `Config`, which is **not** our config, but the configuration for the pool, which took me a sec to get, and then there is **also** `config` the variable, which is now an instance of the pool config and has little to do with our config anymore.

Plus I don't like

```
config.create_pool(...)
```

below, partly also because at first glance I thought "why should our config code know how to create a db connection pool?"

Reply:
Good point, that's indeed confusing. Fixed now, except for `create_pool` which is simply the API of that lib.

Reply:
Yeah sure, nothing we can do there, I guess. Thx!